### PR TITLE
Allow numeric arguments to File.utime.

### PR DESCRIPTION
**Pull request by aardvark179**

### CODE REVIEW

File: src/main/ruby/core/file.rb [hunk 1]
```diff
@@ -1126,8 +1126,10 @@ def self.utime(atime, mtime, *paths)
     end
     paths.each do |path|
       path = Rubinius::Type.coerce_to_path(path)
-      n = POSIX.truffleposix_utimes(path, atime.to_i, atime.usec,
-                                          mtime.to_i, mtime.usec)
+      atime_usec = atime.kind_of?(Time) ? atime.usec : 0;
+      mtime_usec = mtime.kind_of?(Time) ? mtime.usec : 0;
```
Comment (bjfish) [inline src/main/ruby/core/file.rb]: Style: `;`

File: src/main/ruby/core/file.rb [hunk 2]
```diff
@@ -1126,8 +1126,10 @@ def self.utime(atime, mtime, *paths)
...
+      n = POSIX.truffleposix_utimes(path, atime.to_i, atime_usec,
+                                          mtime.to_i, mtime_usec)
```
Comment (eregon) [inline src/main/ruby/core/file.rb]: Should the conversion be done earlier, for example with `Rubinius::Type.coerce_to`?
What does MRI do?

Comment (aardvark179) [reply src/main/ruby/core/file.rb]: Hmm. MRI looks at the `atime` and `mtime` arguments and converts them to a `Time` by effectively using `Time.at` if they aren't already a `Time`. I'll change the code to do that once before looping over the paths. We're still missing some cases that MRI will accept, but that's more a hole in our implementation of `Time.at` and would probably fall over on real file operations anyway.

Comment (eregon) [reply src/main/ruby/core/file.rb]: > We're still missing some cases that MRI will accept

Which cases specifically?

**Pull request merged**